apps/src/code-studio/pd/application_dashboard/detail_view.jsx

```diff
@@ -22,45 +24,81 @@ const renderLineItem = (key, value) => {
   );
 };
 
+class DetailView extends React.Component {
+  static contextTypes = {
+    router: PropTypes.object.isRequired
+  }
+
+  static propTypes = {
+    params: PropTypes.shape({
+      applicationId: PropTypes.string.isRequired
+    }).isRequired
+  }
+
+  state = {
+    loading: true
+  }
+
+  componentWillMount() {
+    this.loadRequest = $.ajax({
+      method: 'GET',
+      url: `/api/v1/pd/applications/${this.props.params.applicationId}`
+    }).done(data => {
+      const formData = JSON.parse(data.form_data);
+      this.setState({
+        data: Object.assign({}, data, {formData: formData}),
+        loading: false
+      });
+    });
+  }
+
+  render() {
+    if (this.state.loading) {
+      return (<Spinner/>);
+    } else {
+      return (
+        (
+          <DetailViewContents
+            applicationData={this.state.data}
+          />
+        )
+      );
+    }
+  }
+}
+
 class DetailViewContents extends React.Component {
```

Comment:
I slightly prefer to keep separate components (at least nontrivial ones) in separate files, and I think Andrew does as well, but it's up to you what makes the most sense in this case.